pylintrc

```diff
@@ -320,7 +320,7 @@ method-naming-style=snake_case
 # Regular expression matching correct method names
 method-rgx=[a-z_][a-z0-9_]{2,}$
 
-# Regular expression which can overwrite the naming style set by typevar-naming-style.
+# Regular expression matching correct type variable names
```

Comment:
This is generated, could you check if there's other occurrence of ``typevar-naming-style`` in the code ? You can then apply it in example files with :
```
pylint --rcfile=/dev/null --generate-rcfile > examples/pylintrc
pylint --rcfile=/dev/null --generate-toml-config > examples/pyproject.toml
```

Reply:
@Pierre-Sassoulas 

> This is generated, could you check if there's other occurrence of typevar-naming-style in the code ?

No other occurrences were found, except for `./doc/whatsnew/fragments/7455.other:1` that was add by this PR.

> You can then apply it in example files with :

With generation commands provided, many extra changes were made.
Plus, `pylintrc` and `examples/pylintrc` also have many differences in `main` branch.

How consistent should these files be?

```plaintext
❯ git diff --stat
 examples/pylintrc       | 291 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------------------------------
 examples/pyproject.toml |   6 ++--
 2 files changed, 150 insertions(+), 147 deletions(-)
```

Reply:
Alright it's only in our own pylintrc that there was an issue then :) Let's merge as is without the fragments then !